packages/liquidator/src/proxyTransactionWrapper.js

```diff
@@ -0,0 +1,288 @@
+    this.gasEstimator = gasEstimator;
+    this.syntheticToken = syntheticToken;
+    this.collateralToken = collateralToken;
+    this.account = account;
+    this.dsProxyManager = dsProxyManager;
+
+    // Helper functions from web3.
+    this.toBN = this.web3.utils.toBN;
+    this.toWei = this.web3.utils.toWei;
+
+    this.isUsingDsProxyToLiquidate = isUsingDsProxyToLiquidate;
+
+    // TODO: refactor the router to pull from a constant file.
+    const defaultConfig = {
+      isUsingDsProxyToLiquidate: {
+        value: false,
+        isValid: x => {
+          return typeof x == "boolean";
+        }
+      },
+      uniswapRouterAddress: {
+        value: "0x7a250d5630b4cf539739df2c5dacb4c659f2488d",
+        isValid: x => {
+          return this.web3.utils.isAddress(x);
+        }
+      },
+      uniswapFactoryAddress: {
+        value: "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f",
+        isValid: x => {
+          return this.web3.utils.isAddress(x);
+        }
+      },
+      liquidatorReserveCurrencyAddress: {
+        value: "",
+        isValid: x => {
+          return this.web3.utils.isAddress(x) || x === "";
+        }
+      },
+      maxReserverTokenSpent: {
+        value: MAX_UINT_VAL,
+        isValid: x => {
+          return typeof x == "string";
+        }
+      }
+    };
+
+    // Validate and set config settings to class state.
+    const configWithDefaults = createObjectFromDefaultProps(proxyTransactionWrapperConfig, defaultConfig);
+    Object.assign(this, configWithDefaults);
+
+    // Preform some basic initalization sanity checks.
+    if (this.isUsingDsProxyToLiquidate) {
+      assert(
+        this.dsProxyManager && this.dsProxyManager.getDSProxyAddress(),
+        "DSProxy Manger has not yet been initialized!"
+      );
+      assert(this.dsProxyManager != undefined, "Cant use dsProxy to liquidate if the client is set to undefined!");
+      assert(
+        this.web3.utils.isAddress(this.liquidatorReserveCurrencyAddress),
+        "Must provide a reserve currency address to use the proxy transaction wrapper!"
+      );
+    }
+
+    this.reserveToken = new web3.eth.Contract(getAbi("ExpandedERC20"), this.liquidatorReserveCurrencyAddress);
+    this.ReserveCurrencyLiquidator = getTruffleContract("ReserveCurrencyLiquidator", web3, "latest");
+  }
+
+  // TODO: wrap this into a common util.
+  createContractObjectFromJson(contractJsonObject) {
+    let truffleContractCreator = truffleContract(contractJsonObject);
+    truffleContractCreator.setProvider(web3.currentProvider);
+    return truffleContractCreator;
+  }
+
+  // Get the effective synthetic token balance. If the bot is executing in normal mode (liquidations sent from an EOA)
+  // then this is simply the token balance of the unlocked account. If the liquidator is using a DSProxy to liquidate,
+  // then consider the synthetics could be minted, + any synthetics the DSProxy already has.
+  async getEffectiveSyntheticTokenBalance() {
```

Comment:
great point and a good refactor. it would be posible to return say the "effective" balance from the smart contract and then call this as a view function. However, this contract is already at the limit of size and any extra returns cause it to bow over the limit :(

For now I think its ok to leave like this, if you ok with that?

Reply:
Yeah, that's cool with me. Are you saying the library smart contract is at the bytecode limit? That's kinda hard to believe....

Reply:
it hits a stack too deep error with any additional state vars or any return types.